src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -91,6 +91,45 @@ public static string GetFullPath(string path)
             return fullPath;
         }
 
+        public static string GetFullPath(string path, string basePath)
+        {
+            if (basePath == null)
+                throw new ArgumentNullException(nameof(basePath));
+
+            if (!IsPathFullyQualified(basePath))
+                throw new ArgumentException(SR.Arg_BasePathNotFullyQualified);
+
+            if (IsPathFullyQualified(path))
+                return GetFullPath(path);
+
+            int length = path.Length;
+
+            if (PathInternal.IsDevice(basePath))
+            {
+                return RemoveRelativeSegments(CombineNoChecks(basePath, path), PathInternal.GetRootLength(basePath));
```

Comment:
Add some comment lines to your cases- this stuff is hard to follow without them.